### PR TITLE
Add support for parentLocales

### PR DESCRIPTION
**Pull request by natemueller**
Use the parentLocales section to figure out the correct fallback
locales.  Can not use customization feature of I18n::Locale::Fallbacks
since it always places the default parent local before any customizations.

### CODE REVIEW

File: lib/cldr/export.rb [hunk 1]
```diff
@@ -119,11 +119,25 @@ def shared_data(component, options = {})
       end
 
       def locales(locale, component, options)
-        locale = locale.to_s.gsub('_', '-').to_sym
+        to_i18n = lambda {|l| l.to_s.gsub('_', '-').to_sym }
```
Comment (camertron) [inline lib/cldr/export.rb]: Why lambdas and not regular 'ol methods?

File: lib/cldr/export.rb [hunk 2]
```diff
@@ -119,11 +119,25 @@ def shared_data(component, options = {})
...
+        from_i18n = lambda {|l| l.to_s.gsub('-', '_') }
+        locale = to_i18n[locale]
 
         locales = if options[:merge]
-          # passing locale itself into constructor as a default fallback to avoid falling back to :en locale
-          I18n::Locale::Fallbacks.new([locale])[locale]
+          defined_parents = Cldr::Export::Data::ParentLocales.new
+
+          ancestory = [locale]
```
Comment (camertron) [inline lib/cldr/export.rb]: Spelling: "ancestry"

**Comment by davispuh** [comment]
I seems like there's some bug

`$ thor cldr:export`

```
ruby-cldr/lib/cldr/export/data/parent_locales.rb:7:in `initialize': wrong number of arguments (given 1, expected 0) (ArgumentError)
from ruby-cldr/lib/cldr/export.rb:95:in `new'
from ruby-cldr/lib/cldr/export.rb:95:in `block in locale_based_data'
from ruby-cldr/lib/cldr/export.rb:94:in `each'
from ruby-cldr/lib/cldr/export.rb:94:in `inject'
from ruby-cldr/lib/cldr/export.rb:94:in `locale_based_data'
from ruby-cldr/lib/cldr/export.rb:88:in `data'
from ruby-cldr/lib/cldr/export/yaml.rb:11:in `export'
from ruby-cldr/lib/cldr/export.rb:70:in `block (2 levels) in export'
from ruby-cldr/lib/cldr/export.rb:69:in `each'
from ruby-cldr/lib/cldr/export.rb:69:in `block in export'
from ruby-cldr/lib/cldr/export.rb:68:in `each'
from ruby-cldr/lib/cldr/export.rb:68:in `export'
from ruby-cldr/lib/cldr/thor.rb:27:in `export'
```


```ruby
def locale_based_data(component, locale, options = {})
        data = locales(locale, component, options).inject({}) do |result, locale|
        data = Data.const_get(component.to_s.camelize).new(locale)
```

it calls `ParentLocales.new(locale)` which doesn't take any argument.

**Comment by camertron** [comment]
@davispuh Oops, my bad! I don't use `thor cldr:export` so I didn't notice it was broken. #40 should fix it :)

**Pull request merged**